### ISSUE TITLE
Service Election is case-sensitive to hostnames - breaking service elections

### ISSUE THREAD

**Issue by simonmicro** [kube-vip/kube-vip#520]
**Describe the bug**
Most Linux distributions allow hostnames to be case-sensitive like "Abc-Abc". Kubernetes enforces valid DNS-names (see https://kubernetes.io/docs/concepts/architecture/nodes/) for node-names and automatically converts them to lower-case (like "abc-abc"). This breaks the service-election of this project, as it compares the `os.Hostname()` (again "Abc-Abc") to the (host-)names of the known endpoints of a service (could be `["abc-abc", "abc-def", ...]`).

Possible solutions (**fat** is recommended):

- Use the `strings.toLower()` method of Go (-> https://www.geeksforgeeks.org/how-to-convert-a-string-in-lower-case-in-golang/) to suppress this problem
- Extend the logging to complain about this
- **Retrieve the own node name from the kubernetes api, instead of using the hostname (because users may join/use a node under another hostname anyways - the Kubernetes node-name will be stable and won't reflect those changes!)**

**To Reproduce**
1. Create a (Debian) Kubernetes node and set their hostname to "Node-ABC"
2. Join the node the cluster - its name will be "node-abc"
3. Try to use a service with the `externalTrafficPolicy` set to `Local`
4. See the `kube-vip`-pod never assigning any IPv4 (on `v0.5.0` error-ing out with no local pods found, even with pods available)

**Expected behavior**
See before.

**Screenshots**
Irrelevant.

**Environment (please complete the following information):**
 - OS/Distro: Debian 11
 - Kubernetes Version: 1.23.2
 - Kube-vip Version: 0.5.0/0.5.10

**`Kube-vip.yaml`:**
Irrelevant.

**Additional context**
_I'm not a Go-developer and I have no idea where the service-election is using the hostname nowadays. I noticed this bug in v0.5.0 and diagnosed it there -> therefore I can't open a PR to fix this on my own._

**Comment by QtSYnT** [comment]
We ran into an issue related to this where our node hostname differed from the k8s node name. So ideally using the actual k8s node name from the API server would be idea.

Or accept it via an env variable so we can do:

```
- name: node_name
  valueFrom:
    fieldRef:
      fieldPath: spec.nodeName
```

**Comment by d-uzlov** [comment]
You may be interested in testing my pull request, which should fix this issue:
- https://github.com/kube-vip/kube-vip/pull/811

**Comment by d-uzlov** [comment]
This should be fixed in `v0.8.0`
